### DISCUSSION TITLE
Search case sensitivity

### DISCUSSION THREAD

**Discussion by discoverlance-com** [rappasoft/laravel-livewire-tables#453] [Q&A] [answered]
I realized after a couple of searches that the search input was not case insensitive, is there any reason for this or any way to work around that. Users searching through a list ideally should be able to enter the value without having to worry about the case sensitivity so I think a search with case sensitivity disabled is much more optimal.

Below is an example of what I am currently trying to search through.

```
# database columns: name, date

#table
public function columns(): array {
  return [
     Column::make('Event Name','name')
     ->sortable()
     ->searchable()
  ];
}

```

**Comment by rappasoft** [comment, marked as answer]
There is nothing forcing the search to be case-sensitive in the package. See [here](https://tables.laravel-boilerplate.com/tailwind?filters%5Bsearch%5D=steph).

**Reply by discoverlance-com** [reply]
I am not sure if then it's because of my code but I am using a similar format as what it says in the docs but for some reason, all my searches are case sensitive. Do you have any ideas as to what could be the possible causes? 
**I am using laravel 8 and postgresql as my database.**

**Reply by rappasoft** [reply]
I'm not a postgres guru but I did do one project using postgres and the tables and I had the same issue. So it's probably a setting in postgres, I ended up overriding the search and using `ILIKE` to make it case insensitive. I'm sure if I researched more there's probably a setting somewhere causing it.

**Reply by discoverlance-com** [reply]
Ok yes, based on what you mentioned, I also did some searching on that and found as you said, it's something with postgres so I think for now will go with the option to override the search with `ILIKE`.

**Reply by acoyco** [reply]
Hello, I know it is a closed discussion, but how did you get override the search with ILIKE? I'm just using searcheable() method on columns.

**Reply by esk-ap** [reply]
now, the solution is to use query callback for column

```
Column::make(__("Desc"), 'description')
                ->collapseAlways()
                ->sortable()
                ->searchable(
            fn(Builder $query, $searchTerm) => $query->orWhere('description',  'ilike', '%' . $searchTerm . '%')
                ),
```

**Comment by duffn** [comment]
I came to discuss the same feature request. I think it would be nice if the package detected PostgreSQL and then used `ilike` in that instance. It certainly works but I'm doing the same, overriding `query()` in all cases only to get case insensitivity.

**Reply by elsoftpy** [reply]
I think so too, it happens with Oracle databases as well, although I don't know wether Oracle has the `ilike`  option

**Reply by lrljoe** [reply]
If this would be useful, please post a Feature Request in a fresh discussion, as it'll help @rappasoft identify any demand for it!

You can set a callback for searchable() by the way, so you can define what "searching" means for that particular field.

**Reply by ChandanShakya** [reply]
> I think so too, it happens with Oracle databases as well, although I don't know wether Oracle has the `ilike` option

did you find any solutions to this?

**Reply by lrljoe** [reply]
@ChandanShakya  - please raise a new discussion for Oracle, and we can look at how we can support it.

**Reply by ChandanShakya** [reply]
I just did it like this
```
Column::make('Name', 'full_name')
  ->searchable(
      fn (Builder $query, $searchTerm) => $query->orWhereRaw('LOWER(full_name) LIKE ?', ['%'.strtolower($searchTerm).'%'])
  )->sortable(),
```

**Reply by lrljoe** [reply]
> I just did it like this
> 
> ```
> Column::make('Name', 'full_name')
>   ->searchable(
>       fn (Builder $query, $searchTerm) => $query->orWhereRaw('LOWER(full_name) LIKE ?', ['%'.strtolower($searchTerm).'%'])
>   )->sortable(),
> ```

Yep, if you open up a new discussion, then we can add in support, rather than using "raw" statements.

Feel free to prod me on the Discord channel to discuss tho!

**Reply by ChandanShakya** [reply]
> Yep, if you open up a new discussion, then we can add in support, rather than using "raw" statements.

Opened discussion for Oracle support.
https://github.com/rappasoft/laravel-livewire-tables/discussions/2252
